HopsanCore/src/Parameters.cpp

```diff
@@ -753,7 +747,10 @@ bool ParameterEvaluatorHandler::refreshParameterValueText(const HString &rParame
 bool ParameterEvaluatorHandler::evaluateInSystemParent(const HString &rName, HString &rEvaluatedParameterValue, const HString &rType)
 {
     // Try one of our system parent component parameters
-    if(mParentComponent && mParentComponent->getSystemParent())
+    if(mParentComponent && mParentComponent->isComponentSystem()) {
```

Comment:
I am not sure this code does what you would think when reading the variable names, or possibly it did the wrong thing before, or possible your changes require this new behavior. 

Does the following make sense:  The Parameter evaluator is attached to a component, this is the badly named mParentComponent (I am fairly sure about this, but not completely).

evaluateInSystemParent, should evaluate only in the system that the current component is inside, in its system-parent. This is what the original code did. But now it first checks if the attached component (badly named mParentComponent) is itself a SystemComponent, and if that is the case, it evaluates the parameter locally. And this would be the same as self.paramname, so this may not work in case the actual component is a system. This could happen with programmed subsystems, or maybe from NumHop script evaluation.

But we should really write some kind of tests for this to make sure. I will try to make one.
But right now I think that the above code is not doing what it is supposed to be doing.